### PR TITLE
Ensure timestamp is always naive datetime for comparison

### PR DESCRIPTION
**Pull request by gcapalbo**
@emord @benrudolph

### CODE REVIEW

File: corehq/apps/data_interfaces/models.py [hunk 1]
```diff
@@ -217,6 +218,16 @@ def _try_date_conversion(self, date_or_string):
 
         return date_or_string
 
+    def clean_datetime(self, timestamp):
+        if not isinstance(timestamp, datetime):
+            timestamp = datetime.combine(timestamp, time(0, 0))
+
+        if timestamp.tzinfo:
+            # Convert to UTC and make it a naive datetime for comparison to datetime.utcnow()
+            timestamp = timestamp.astimezone(pytz.utc).replace(tzinfo=None)
```
Comment (benrudolph) [inline corehq/apps/data_interfaces/models.py]: I think you could just use: https://github.com/dimagi/dimagi-utils/blob/master/dimagi/utils/parsing.py#L36

**Review by benrudolph — APPROVED** [review]
one comment that might be useful, but it's fairly minor. will let you decide

File: corehq/apps/data_interfaces/models.py (hunk 1 above)
Comment (gcapalbo) [inline corehq/apps/data_interfaces/models.py]: hmm, looks like that takes a string though and at this point i've already got a timezone-aware datetime

Comment (benrudolph) [inline corehq/apps/data_interfaces/models.py]: ah oops missed that. looks good then

**Pull request merged**